src/lcals/FIRST_MIN-Cuda.cpp

```diff
@@ -80,24 +80,35 @@ void FIRST_MIN::runCudaVariantImpl(VariantID vid)
 
        FIRST_MIN_MINLOC_INIT;
 
+       const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
        MyMinLoc* dminloc;
-       cudaErrchk( cudaMalloc( (void**)&dminloc, sizeof(MyMinLoc) ) );
-       cudaErrchk( cudaMemcpy( dminloc, &mymin, sizeof(MyMinLoc),
-                               cudaMemcpyHostToDevice ) );
+       MyMinLoc mymin_block[grid_size]; //per-block min value
+       Index_type minloc = m_initloc;
+
+       for (Index_type i=0;i<static_cast<Index_type>(grid_size);i++){
+	       mymin_block[i] = mymin;
+       }
+       cudaErrchk( cudaMalloc( (void**)&dminloc, grid_size * sizeof(MyMinLoc) ) );
+       cudaErrchk( cudaMemcpy( dminloc, mymin_block, grid_size*sizeof(MyMinLoc),
+                               cudaMemcpyHostToDevice ) );      
 
-       const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
        first_min<block_size><<<grid_size, block_size,
                    sizeof(MyMinLoc)*block_size>>>( x,
                                                    dminloc,
                                                    iend );
-       cudaErrchk( cudaGetLastError() );
+	    
+       cudaErrchk( cudaMemcpy( &mymin_block, dminloc, grid_size * sizeof(MyMinLoc),
+                              cudaMemcpyDeviceToHost ) );       
+
+       for (Index_type i=0;i<static_cast<Index_type>(grid_size);i++){
+           if( (mymin_block[i]).val < mymin.val)
+                   mymin = mymin_block[i];
+       }
 
-       cudaErrchk( cudaMemcpy( &mymin, dminloc, sizeof(MyMinLoc),
-                               cudaMemcpyDeviceToHost ) );
        m_minloc = RAJA_MAX(m_minloc, mymin.loc);
```

Comment:
@rhornung67 shouldn't this just be an assignment? What if it was already right and the result was smaller and wrong, then it would not overwrite the existing value and we'd think it was right when it was actually wrong.

Reply:
I wouldn't change it here, but I'm not sure this is quite right.

Reply:
I think it can be an assignment since MyMinLoc is a POD type: https://github.com/LLNL/RAJAPerf/blob/develop/src/lcals/FIRST_MIN.hpp#L41

Reply:
Updated the comparison method

Reply:
> @rhornung67 shouldn't this just be an assignment? What if it was already right and the result was smaller and wrong, then it would not overwrite the existing value and we'd think it was right when it was actually wrong.

@MrBurmark I mis-interpreted your question when you asked it originally. Now, I see what you were asking. Yes, that can be an assignment and probably should be to make the code more clear. I'll push a change.

Reply:
This looks good.